mypy/errors.py

```diff
@@ -180,9 +164,7 @@ def copy(self) -> 'Errors':
         new.type_name = self.type_name[:]
         new.function_or_member = self.function_or_member[:]
         new.target_module = self.target_module
-        new.target_class = self.target_class
-        new.target_function = self.target_function
-        new.target_ignore_depth = self.target_ignore_depth
+        new.scope = self.scope
```

Comment:
I think it makes more sense to have a reference, since making a copy means that using the scope that was passed in doesn't work right any more.

Both are a little dodgy in different situations, though.

Fortunately it seems to not actually matter, as copy only is used while checking expressions, so there won't be any scope changes.